### PR TITLE
Add go race detector config

### PR DESCRIPTION
**Pull request by keliramu**

### CODE REVIEW

File: ci/compile.sh [hunk 1]
```diff
@@ -7,10 +7,8 @@ source "${WORKDIR}"/ci/archs.sh
 # Since race detector has huge performance price and it works only on amd64 and does not
 # work with pie executables, its enabled only for development builds.
 # shellcheck disable=SC2153
-[ "${ENVIRONMENT}" = "dev" ] && [ "${ARCH}" = "amd64" ] && BUILDMODE="-race" || BUILDMODE="-buildmode=pie"
-
-#TODO/FIXME: just to verify, to be removed
-BUILDMODE="-buildmode=pie"
+[ "${ENVIRONMENT}" = "dev" ] && [ "${ARCH}" = "amd64" ] && [ "${RACE_DETECTOR_ENABLED:-""}" != "" ] \
+		&& BUILDMODE="-race" || BUILDMODE="-buildmode=pie"
```
Comment (Savolro) [inline ci/compile.sh:11]: Position independent executable will make debuggers not work by default, which is not something we want

File: ci/compile.sh [hunk 2]
```diff
@@ -7,10 +7,8 @@ source "${WORKDIR}"/ci/archs.sh
 # Since race detector has huge performance price and it works only on amd64 and does not
 # work with pie executables, its enabled only for development builds.
 # shellcheck disable=SC2153
-[ "${ENVIRONMENT}" = "dev" ] && [ "${ARCH}" = "amd64" ] && BUILDMODE="-race" || BUILDMODE="-buildmode=pie"
-
-#TODO/FIXME: just to verify, to be removed
-BUILDMODE="-buildmode=pie"
+[ "${ENVIRONMENT}" = "dev" ] && [ "${ARCH}" = "amd64" ] && [ "${RACE_DETECTOR_ENABLED:-""}" != "" ] \
```
Comment (Savolro) [inline ci/compile.sh:10]: Position independent executable will make debuggers not work by default, which is not something we want for dev builds.

File: ci/compile.sh (hunk 1 above)
Comment (Savolro) [inline ci/compile.sh:11]: Position independent executable will make debuggers not work by default, which is not something we want for dev builds.

**Review by Savolro — CHANGES_REQUESTED** [review]
`

File: ci/compile.sh (hunk 1 above)
Comment (keliramu) [reply ci/compile.sh:11]: For local build developer should add new variable to `.env` file. Need sample?

Comment (keliramu) [reply ci/compile.sh]: Fixed

**Review by bartoszWojciechO — APPROVED** [review]

**Pull request merged**